tests/unit/_modules/test_proposal.py

```diff
@@ -129,3 +146,142 @@ def test_create(self, output_helper):
         assert len(prop['nvme-ssd-spinner']) is 0
         assert len(prop['nvme-ssd']) is p.DEFAULT_DATA_R
         assert len(prop['nvme-spinner']) is p.DEFAULT_DATA_R
+
+
+    def replace_disk(self, type='hdd', ratio=2):
+        # idx 0-11 -> HDD
+        # idx 12-17 -> SSD
+        # idx 18 -> NVME
+        wal_db_replaced = None
+        hwinfo_out = OutputHelper().cephdisks_output
+        p = proposal.Proposal(OutputHelper().cephdisks_output, ratio=ratio)
+        old_prop = p.create()
+
+        # replace a HDD
+        hwinfo_out_new = hwinfo_out
```

Comment:
this is a bit misleading, `hwinfo_out_new` is not a copy `hwinfo_out`, just a second reference to the same list. It does not matter here because `hwinfo_out` is only used for the max index when inserting the new disk at a random spot, but can be confusing.

Since only one `hwinfo_out` is needed, we don't need to create a second reference. If we want to make the code easier to understand by having a second list, we can use list slicing to copy all values: `hwinfo_out_new = hwinfo_out[:]`